src/main/game/Player.java

```diff
@@ -89,42 +88,53 @@ public void addColonist(Colonist colonist){
         colonists.add(colonist);
     }
 
-    /*
-    public void addWorker(Worker worker){
-        workers.add(worker);
-    }
-    */
 
-    public void removeMelee(Melee melee){
-//        for(int i = 0;i<melees.size();i++){
-//            if(melees.get(i).getInstanceId() == melee.getInstanceId()){
-//                melees.remove(i);
-//            }
-//        }
-    }
+    public void addWorker(Worker worker){workers.add(worker);}
 
-    public void removeRanged(Ranged ranged){
-//        for(int i = 0;i<ranges.size();i++){
-//            if(ranges.get(i).getInstanceId() == ranges.getInstanceId()){
-//                ranges.remove(i);
-//            }
-//        }
-    }
 
-    public void removeExplorer(Explorer explorer){
-//        for(int i = 0;i<explorers.size();i++){
-//            if(explorers.get(i).getInstanceId() == explorer.getInstanceId()){
-//                explorers.remove(i);
-//            }
-//        }
-    }
+    //Loops through all the arraylists to remove entities base on entity id
+    public void removeEntity(EntityId entityId){
+        for(int i =0;i<melees.size();i++){
```

Comment:
This shouldn't have to loop through every single unit. Can't you look at the part of the entityID that determines the type to see what type to loop through.